src/Apps/W1/External File Storage - SharePoint Connector/App/src/ExtSharePointConnectorImpl.Codeunit.al

```diff
@@ -29,29 +31,11 @@ codeunit 4580 "Ext. SharePoint Connector Impl" implements "External File Storage
     /// <param name="FilePaginationData">Defines the pagination data.</param>
     /// <param name="TempFileAccountContent">A list with all files stored in the path.</param>
     procedure ListFiles(AccountId: Guid; Path: Text; FilePaginationData: Codeunit "File Pagination Data"; var TempFileAccountContent: Record "File Account Content" temporary)
-    var
-        SharePointFile: Record "SharePoint File";
-        SharePointClient: Codeunit "SharePoint Client";
-        OrginalPath: Text;
     begin
-        OrginalPath := Path;
-        InitPath(AccountId, Path);
-        InitSharePointClient(AccountId, SharePointClient);
-        if not SharePointClient.GetFolderFilesByServerRelativeUrl(Path, SharePointFile) then
-            ShowError(SharePointClient);
-
-        FilePaginationData.SetEndOfListing(true);
-
-        if not SharePointFile.FindSet() then
-            exit;
-
-        repeat
-            TempFileAccountContent.Init();
-            TempFileAccountContent.Name := SharePointFile.Name;
-            TempFileAccountContent.Type := TempFileAccountContent.Type::"File";
-            TempFileAccountContent."Parent Directory" := CopyStr(OrginalPath, 1, MaxStrLen(TempFileAccountContent."Parent Directory"));
-            TempFileAccountContent.Insert();
-        until SharePointFile.Next() = 0;
+        if GetUseGraphAPI(AccountId) then
```

Comment:
I generally like the refactoring to the helper codeunits, but we now will always have to Get Operations.
The record should be cached, but I'm still not sure if this is good.

I would suggest to change the helper methods to accept the SharePointAccount table as parameter instead.